### PR TITLE
Any Sequencer implements

### PR DESCRIPTION
**Pull request by jinchihe**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**

There is a requirement:  When any one of the Job1, job2 or job3 complete successfully, Job4 will be started. Order doesn’t matter, but we don’t want to wait for all the job status.

 
image

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by fenglixa** [comment]
@jinchihe 
Some Lint issues your code.
```
import kfp
^
sdk/python/tests/compiler/testdata/any_sequencer_sample.py:2:1: F401 'kfp.components' imported but unused
from kfp import components
^
1     E902 IndentationError: unindent does not match any outer indentation level
2     F401 'kfp' imported but unused
Makefile:78: recipe for target 'lint' failed
make: *** [lint] Error 1
```

You can run `make lint` `make check_license` `make unit_test` to see detail issues before commit code

**Comment by jinchihe** [comment]
@fenglixa @Tomcli The CI test passed, ready for another look. Thanks.

File: sdk/python/tests/compiler/testdata/any_sequencer.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      taskSpec:
+        params:
+        - {name: pipelineRun-namespace}
+        - {name: pipelineRun-name}
+        steps:
+        - name: main
+          args: [-namespace, $(params.pipelineRun-namespace), -prName, $(params.pipelineRun-name),
+            -taskList, 'task1,task2,task3']
+          command: [any-taskrun]
+          image: dspipelines/any-sequencer:latest
```
Comment (Tomcli) [inline sdk/python/tests/compiler/testdata/any_sequencer.yaml:79]: thanks @jinchihe, are you able to provide the source code for this `dspipelines/any-sequencer` image?

**Comment by drewbutlerbb4** [comment]
Thanks @jinchihe!

This will require a UI update because the UI doesn't account for the `anyConditions` dependencies in the DAG. I will submit a PR to cover that after this PR is merged.

**Comment by fenglixa** [comment]
> This will require a UI update because the UI doesn't account for the `anyConditions` dependencies in the DAG. I will submit a PR to cover that after this PR is merged.

Here is the UI PR #401. Thanks @drewbutlerbb4

File: sdk/python/tests/compiler/testdata/any_sequencer.yaml (hunk 1 above)
Comment (jinchihe) [reply sdk/python/tests/compiler/testdata/any_sequencer.yaml:79]: Already commit in the PR, thanks @Tomcli

**Comment by jinchihe** [comment]
@Tomcli @fenglixa The image related code committed, thanks.
/cc @vincent-pli

File: sdk/python/tests/compiler/testdata/any_sequencer.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        command=["/bin/bash", "-c"],
+        arguments=["sleep 300"]
+    )
+
+    task4 = dsl.ContainerOp(
+        name="task4",
+        image="registry.access.redhat.com/ubi8/ubi-minimal",
+        command=["/bin/bash", "-c"],
+        arguments=["sleep 30"]
+    ).apply(after_any([task1, task2, task3]))
```
Comment (Tomcli) [inline sdk/python/tests/compiler/testdata/any_sequencer.py:52]: Quick question: Is there a limit for how many tasks the `Any Sequencer` can handle?

**Review by Tomcli — COMMENTED** [review]
/lgtm

File: sdk/python/tests/compiler/testdata/any_sequencer.py (hunk 2 above)
Comment (jinchihe) [reply sdk/python/tests/compiler/testdata/any_sequencer.py:52]: No limitation for task numbers. Thanks.

**Comment by jinchihe** [comment]
@fenglixa any more comments? if not, would you please approve the ticket? Thanks.

**Pull request merged**